roles/generate-docs/tasks/main.yml

```diff
@@ -0,0 +1,7 @@
+---
+# tasks file for gen-docs
+
+- name: Register template
```

Comment:
Maybe just name this "Generate Documentation based on provided template and facts"?